src/main/java/appeng/services/compass/CompassRegion.java

```diff
@@ -85,18 +79,20 @@ private void openFile( boolean create )
 
 	private File getFileName()
 	{
-		String folder = this.rootFolder.getPath() + File.separatorChar + "compass";
-		File folderFile = new File( folder );
+		assert this.rootFolder.isDirectory();
```

Comment:
Not sure about the assert here. These are usually disabled and thus not really providing any information outside of indev. Maybe raise as `PreCondition` to the constructor?